qiita_db/metadata_template/sample_template.py

```diff
@@ -162,6 +162,26 @@ def columns_restrictions(self):
         """
         return qdb.metadata_template.constants.SAMPLE_TEMPLATE_COLUMNS
 
+    def delete_sample(self, sample_name):
+        """Delete `sample_name` from sample information file
+
+        Parameters
+        ----------
+        sample_name : str
+            The sample name to be deleted
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the `sample_name` has been used in a prep info file
+        """
+        if any([pt.get(sample_name) != None
+                for pt in qdb.study.Study(self.study_id).prep_templates()]):
+            raise qdb.exceptions.QiitaDBColumnError(
+                "'%s' has been linked in a prep template" % (sample_name))
```

Comment:
It would perhaps be handy to know what prep templates it has been linked to, for troubleshooting?